rhel/vex/parser.go

```diff
@@ -290,8 +272,47 @@ func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulne
 			return nil, fmt.Errorf("could not unbind cpe: %s %w", ch, err)
 		}
 		vuln.Repo = c.rc.Get(wfn)
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			if err != nil {
+				zlog.Warn(ctx).
+					Str("purlHelper", purlHelper).
+					Err(err).
+					Msg("could not parse PURL")
+			}
+			if !checkPURL(purl) {
+				continue
```

Comment:
I'm not sure I fully understand this, we need to extract the pURL before we can check it. Maybe we can move it before the creation of the vulnerability, but I think it's a little more readable setting the `vuln.Repo` and `vuln.Range` directly rather than dealing with more variables